pkg/autodetect/main.go

```diff
@@ -147,6 +154,41 @@ func (b *Background) detectClusterRoles() {
 	}
 }
 
+func (b *Background) cleanDeployments() error {
+	log.Info("Detecting orphaned deployments.")
+	deployments := &appsv1.DeploymentList{}
+	opts := &client.ListOptions{}
+	instances := &v1.JaegerList{}
+
+	// get all deployments
+	if err := b.cl.List(context.Background(), opts, deployments); err != nil {
+		return err
+	}
+
+	// get all jaeger instances
+	if err := b.cl.List(context.Background(), opts, instances); err != nil {
```

Comment:
Are we able to get *only* the deployment name+namespace and the annotations, instead of the full objects? I'm afraid we could be causing unnecessary load on the cluster by retrieving all *deployments* from all namespaces that this operator has access to.

@jkandasa, @kevinearls: could you add a case for your test plans? It would be good to test how the operator behaves when there are 100/1.000/10.000 deployments in the cluster.

Reply:
@jpkrohling Sure, a test case would be great.  The best thing to do would be create a ticket and assign it to me.  That way I'll be sure not to lose track of it...

Reply:
@jpkrohling I haven't found a way to get only the name and annotations, I did some modifications, I'm using LabelSelector to get the deployments already labeled by the operator.

Reply:
@jpkrohling I haven't found a way to get only the name and annotations, I did some modifications, I'm using LabelSelector to get the deployments already labeled by the operator. That could help to mitigate performance issues.

Reply:
#584 created.